### ISSUE TITLE
Add support for additional Azure environment endpoints

### ISSUE THREAD

**Issue by anweiss** [Azure/azure-sdk-for-go#40]
Should include support for additional Service Management endpoints for both Azure Government and Azure China. Constants block should include those endpoints.

```go
const (
	paramNotSpecifiedError = "Parameter %s is not specified."

	azureManagementDnsName           = "https://management.core.windows.net"
	azureGovernmentManagementDnsName = "https://management.core.us-govcloudapi.net"
	azureChinaManagementDnsName      = "https://management.core.chinacloudapi.cn"
	msVersionHeader                  = "x-ms-version"
	msVersionHeaderValue             = "2014-05-01"
	contentHeader                    = "Content-Type"
	defaultContentHeaderValue        = "application/xml"
	requestIdHeader                  = "X-Ms-Request-Id"
)
```

Will be best to wait until #39 is merged before making this add. The `NewClient` functions could include some sort of `Environment` argument to denote which environment should be used.

**Comment by ahmetalpbalkan** [comment]
This should be configurable rather than having China also hardcoded, that's the practice I followed in blob storage package and what other SDKs do. 

Public cloud is hardcoded by default and users can provide their own base URLs.

**Comment by anweiss** [comment]
Agreed.

**Comment by ahmetalpbalkan** [comment]
@anweiss wanna make a PR after we merge #39?

**Comment by anweiss** [comment]
Sure thing!

**Comment by ahmetalpbalkan** [comment]
closed with #48
